### ISSUE TITLE
Gather 2018 Distinct Value lists

### ISSUE THREAD

**Issue by tucotuco** [VertNet/dwc-qa-manage#39]
Our distinct value lists from 2017 are more than a year old now. We intended to try to make annual copies of these, so any time now will be good to gather these again.

John can do this for VertNet and request it of GBIF.

**Comment by debpaul** [comment]
Thanks John - I’ll reach out to iDigBio again and see if ALA will join us now.

Sent from Shoe (my iPhone)

On Jun 4, 2018, at 2:04 PM, John Wieczorek <notifications@github.com<mailto:notifications@github.com>> wrote:


Our distinct value lists from 2017 are more than a year old now. We intended to try to make annual copies of these, so any time now will be good to gather these again.

John can do this for VertNet and request it of GBIF.

**Comment by Tasilee** [comment]
Thanks Deb: I have notified our lead ALA developer Nick Dosremedios about this.

**Comment by tucotuco** [comment]
I have made the request to Tim Robertson at GBIF who put them together for us last time. I'll work on the VertNet values.

**Comment by nickdos** [comment]
I'm happy to generate such lists from the ALA. Is there is set of DwC fields (and/or non-DwC fields) that I can use to query with?

**Comment by tucotuco** [comment]
Thanks, Nick. We have been trying to gather distinct value lists for terms
(with count) for Occurrences that might benefit from controlled
vocabularies. Here is a list of what others have been summarizing:

basisOfRecord
continent
countrycode
country
day
disposition
establishmentMeans
geodeticDatum
georeferenceVerificationStatus
identificationQualifier
identificationVerificationStatus
islandGroup
island
language
license
lifeStage
month
nomenclaturalCode
occurrenceStatus
organismScope
preparations
reproductiveCondition
sex
taxonRank
taxonomicStatus
typeStatus
type
verbatimSRS
waterbody

It looks like iDigBio also added some indexed versions of terms for
comparisons of interest (
https://github.com/tdwg/dwc-qa/tree/master/data/idigbioDistinctValues).

And here is an example csv from last year from VertNet for basisOfRecord
with header to include DwC term name and "reps" as the number of
Occurrences it appeared in:

https://github.com/tdwg/dwc-qa/blob/master/data/VNDistinctValues/VertNet_distinct_basisOfRecord_2017-02-14.csv

**Comment by debpaul** [comment]
Yes please Nick, if you are also willing/able to share the indexes versions - that would be great. These are super useful for helping people to understand indexing...(and more).

Excited to have you on board. Thank you.

Sent from Shoe (my iPhone)

On Jun 5, 2018, at 8:41 AM, John Wieczorek <notifications@github.com<mailto:notifications@github.com>> wrote:

Thanks, Nick. We have been trying to gather distinct value lists for terms
(with count) for Occurrences that might benefit from controlled
vocabularies. Here is a list of what others have been summarizing:

basisOfRecord
continent
countrycode
country
day
disposition
establishmentMeans
geodeticDatum
georeferenceVerificationStatus
identificationQualifier
identificationVerificationStatus
islandGroup
island
language
license
lifeStage
month
nomenclaturalCode
occurrenceStatus
organismScope
preparations
reproductiveCondition
sex
taxonRank
taxonomicStatus
typeStatus
type
verbatimSRS
waterbody

It looks like iDigBio also added some indexed versions of terms for
comparisons of interest (
https://github.com/tdwg/dwc-qa/tree/master/data/idigbioDistinctValues).

And here is an example csv from last year from VertNet for basisOfRecord
with header to include DwC term name and "reps" as the number of
Occurrences it appeared in:

https://github.com/tdwg/dwc-qa/blob/master/data/VNDistinctValues/VertNet_distinct_basisOfRecord_2017-02-14.csv

**Comment by tucotuco** [comment]
It might also be interesting for all of us to add distinct values for the
year term.

**Comment by debpaul** [comment]
That would be most useful, instructive, and entertaining
Deb

Sent from Shoe (my iPhone)

On Jun 5, 2018, at 9:06 AM, John Wieczorek <notifications@github.com<mailto:notifications@github.com>> wrote:

It might also be interesting for all of us to add distinct values for the
year term.

**Comment by tucotuco** [comment]
VertNet distinct values added in commit https://github.com/tdwg/dwc-qa/commit/449824b992c74a351e94b3f4d4b6330fb5711e86.

**Comment by nickdos** [comment]
I've managed to pull out unique values for a subset of fields from the ALA SOLR index. We don't index all fields, so the missing fields might be able to be generated via a Cassandra (I don't know how to). I figured this subset would be a good start and our next major release should include all DwC fields (we're moving to a clustered architecture to handle the bigger data).

Should I attach the TXT file to this issue or commit it to a directory or another repo - I noticed the comment above references a commit that is not linked in this repo, so wanted to check first.

Edit: [ZIP file with shell script and output from script](https://www.dropbox.com/s/mla5fkjf6ke0g7j/ALA_DwC_distinct_values_20180704.zip?dl=0)

fields used: `basis_of_record country_code country month year establishment_means raw_identification_qualifier license occurrence_status_s reproductive_condition_s raw_sex rank type_status`

**Comment by tucotuco** [comment]
Hi Nick, That's great. If you clone or fork the tdwg/dwc-qa repository,
create a new branch, add a folder for ALA, add the files to that folder,
commit, push and make a pull request, that would be ideal.

**Comment by nickdos** [comment]
Hi @tucotuco, I've created another PR with some changes, including the suggested readme file, using sub-directories with date, as well as indicating "index" values in the file name, similar to how iDigBio does it.
